include/zephyr/storage/flash_map.h

```diff
@@ -269,16 +271,72 @@ uint8_t flash_area_erased_val(const struct flash_area *fa);
 #define FLASH_AREA_SIZE(label) \
 	DT_REG_SIZE(DT_NODE_BY_FIXED_PARTITION_LABEL(label))
 
+/**
+ * Returns DTS node for label, if it is fixed-partition compatible
+ *
+ * @return DTS node with given label, if exists and is a child of
+ *	   a fixed-partition compatible.
+ */
+#define FIXED_PARTITION_NODE(label) \
+	COND_CODE_1(FIXED_PARTITION_EXISTS(DT_NODELABEL(label)), (DT_NODELABEL(label)), ())
+
+/**
+ * Returns non-0 value if fixed-partition of given DTS node label exists.
+ *
+ * @param label DTS node label
+ *
+ * @return non-0 if fixed-partition node exists and is enabled;
+ *	   0 if node does not exist, is not enabled or is not fixed-partition.
+ */
+#define FIXED_PARTITION_EXISTS(label) DT_FIXED_PARTITION_EXISTS(DT_NODELABEL(label))
```

Comment:
It looks like this patch is using DT_FIXED_PARTITION_EXISTS before it is defined in the PR, breaking bisectability. This should be cleaned up so the new macro is added before it is used.

Reply:
Will change the order.
I have implemented everything in flash_map, before deciding to move the body of macro to fixed-partition.h, that is why the order is wrong now.